File: java-playground/src/main/java/org/gonza/javaplayground/util/NumberGenerator.java
package org.gonza.javaplayground.util;

import java.util.Random;

public class NumberGenerator {
    private NumberGenerator() throws InstantiationException {
        throw new InstantiationException("유틸리티 클래스는 인스턴스화 할 수 없습니다.");
    }

    public static int generateRandomNumber() {
        return new Random().nextInt(10);
    }
}

File: kotlin-playground/src/main/kotlin/org/gonza/kotlinplayground/utils/NumberGenerator.kt
package org.gonza.kotlinplayground.utils

import java.util.Random

open class NumberGenerator: TargetGenerator<Int, Int> {

    override fun generate(target: Int): Int {
        val random = Random()
        val number = random.nextInt(target)

        return number
    }
}

class TestNumberGenerator(private val fixedNumber: Int): NumberGenerator() {
    override fun generate(target: Int): Int = fixedNumber
}